common-lib/src/main/java/com/google/cloud/tools/intellij/analytics/UsageTracker.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2016 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.analytics;
+
+import com.google.cloud.metrics.Event;
+import com.google.cloud.metrics.MetricsException;
+import com.google.cloud.metrics.MetricsSender;
+import com.google.cloud.tools.intellij.flags.PropertiesFileFlagReader;
+import com.google.common.collect.ImmutableMap;
+
+import com.intellij.ide.util.PropertiesComponent;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.updateSettings.impl.UpdateChecker;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Map;
+
+public class UsageTracker {
```

Comment:
usage tracking should be a service as in its current implementation. It operates by side-effect and needs to be mocked out for unit testing.